crates/server/src/server/server_future.rs

```diff
@@ -281,7 +284,7 @@ impl<T: RequestHandler> ServerFuture<T> {
     pub fn register_tls_listener(
         &mut self,
         listener: net::TcpListener,
-        timeout: Duration,
+        timeout_duration: Duration,
```

Comment:
I don't love the type name suffix (but I understand the need to avoid shadowing `timeout()`). Maybe `handshake_timeout` or `accept_timeout`?